src/libraries/System.Private.CoreLib/src/System/Decimal.DecCalc.cs

```diff
@@ -192,29 +192,70 @@ private static void UInt64x64To128(ulong a, ulong b, ref DecCalc result)
             /// <param name="bufNum">96-bit dividend as array of uints, least-sig first</param>
             /// <param name="den">32-bit divisor</param>
             /// <returns>Returns remainder. Quotient overwrites dividend.</returns>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             private static uint Div96By32(ref Buf12 bufNum, uint den)
             {
-                // TODO: https://github.com/dotnet/runtime/issues/5213
-                ulong tmp, div;
-                if (bufNum.U2 != 0)
+                if (X86.X86Base.X64.IsSupported)
```

Comment:
This is a lot of extra code for a relatively small performance increase on modern CPUs.

We currently use `Skylake` as the baseline for a lot of our perf score numbers and 3ns savings for 13 lines of new code (+23 more for x86) doesn't really seem worth it.

Ideally any improvements would be shared across all 3 platforms or be significant enough to make the additional complexity worthwhile.

Reply:
Ok I removed the x64 specific code and just kept the x86 since it was just a few ns faster than the x86 specific code (as long **as there are no branch misspredictions**) and the x86 should be faster on `Skylake` even with a misspredicted branch. 

I can change it to x64 only code insted (2 if statements and 2 DivRem calls) if you rather like it.